ngmix/flags.py

```diff
@@ -77,7 +79,7 @@ def get_flags_str(val, name_map=None):
     nstrs = []
     for pow in range(32):
         fval = 2**pow
-        if ((val & fval) != 0):
+        if ((np.uint32(val) & fval) != 0):
```

Comment:
With `numpy>=2`, this now results in an OverflowError. We could either adopt limiting `pow` to `range(30)` as we did in [metadetect here](https://github.com/esheldon/metadetect/pull/224). I figured it's probably safer to cast it instead since that leaves the most-significant bits still usable.

Reply:
We should 

- limit the loop to 31
- raise a warning if the input value is bigger than 2**32-1 or < 0
- safely cast using `np.array(value).astype(dtype)`

A direct call to `np.uint32` is raising a warning that it will raise an error in the future for my local copy of numpy at 1.26.4.

Reply:
Oh, I didn't know there were going to be future releases of `numpy 1.26` that would convert this to an error.

Reply:
They might mean in numpy 2. IDK.